power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -300,8 +278,9 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
         }
     }
 
+    // helper for row or columnar-based components
     template <class CompType, cache_type_c CacheType>
-    void update_component(ConstDataset const& update_data, Idx pos, std::span<Idx2D const> sequence_idx) {
+    void update_component_row_col(ConstDataset const& update_data, Idx pos, std::span<Idx2D const> sequence_idx) {
```

Comment:
Why is this necessary?

Reply:
It isn't. I was trying to remove so many overloads, hence I "added" `update_components` and this one. Since the whole purpose of this one is only to encapsulate the logic where one updates a columnar or a row-based component, I thought it was a good naming convention.

Do you think is more readable if we just change it back to `update_component`? Different name maybe? Or do you have something else in mind?

Reply:
> Do you think is more readable if we just change it back to `update_component`? 

I would argue changing it back but for a different reason: having it specified as `*_row_col` indicates that there are other counterparts. I think if you put some comment to explain the overload functions, that would already help a lot. E.g., "this is the entry point for update all sort of components in update data", "this is the overload to update ***", etc.